content/en/blog/_posts/2022-11-XX-runtime-observability/index.md

```diff
@@ -0,0 +1,275 @@
+working on continuously improving them, for example by [attaching the logs to the
+traces][pr-6294] or extending the [spans to logical parts of the
+application][pr-6343]. This helps users of the traces to gain the same
+information like parsing the logs, but with enhanced capabilities of scoping and
+filtering to other OTEL signals. The CRI-O maintainers are also working on a
+container monitoring replacement for [conmon][conmon], which is called
+[conmon-rs][conmon-rs] and is purely written in [Rust][rust]. One benefit of
+having a Rust implementation is to be able to add features like OpenTelemetry
+support, because the crates (libraries) for those already exist. This allows a
+tight integration with CRI-O and lets consumers see the most low level tracing
+data from their containers.
+
+[pr-4883]: https://github.com/cri-o/cri-o/pull/4883
+[pr-6294]: https://github.com/cri-o/cri-o/pull/6294
+[pr-6343]: https://github.com/cri-o/cri-o/pull/6343
+[conmon]: https://github.com/containers/conmon
+[conmon-rs]: https://github.com/containers/conmon-rs
+[rust]: https://www.rust-lang.org
+
+The [containerd][containerd] folks added tracing support since v1.6.0, which is
+available [by using a plugin][containerd-docs]. Lower level OCI runtimes like
+[runc][runc] or [crun][crun] feature no support for OTEL at all and it does not
+seem to exist a plan for that. We always have to consider that there is a
+performance overhead when collecting the traces as well as exporting them to a
+data sink. I still think it would be worth an evaluation on how extended
+telemetry collection could look like in OCI runtimes. Let's see if the Rust OCI
+runtime [youki][youki-1348] is considering something like that in the future.
+
+[containerd-docs]: https://github.com/containerd/containerd/blob/7def13d/docs/tracing.md
+[youki-1348]: https://github.com/containers/youki/issues/1348
+
+Let's finally give it a try! For my demo I'll stick to a single local node runc,
+conmon-rs, CRI-O, kubelet stack. To enable tracing in the kubelet, we need to
+apply the following `KubeletConfiguration`:
+
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+featureGates:
+  KubeletTracing: true
+tracing:
+  samplingRatePerMillion: 1000000
+```
+
+A `samplingRatePerMillion` equally to one million will internally translate to
+sampling everything. A similar configuration has to be applied to CRI-O, we can
+wither start the `crio` binary with `--enable-tracing` and
+`--tracing-sampling-rate-per-million 1000000` or we use a drop-in configuration
+like this:
+
+```shell
+> cat /etc/crio/crio.conf.d/99-tracing.conf
+```
+
+```toml
+[crio.tracing]
+enable_tracing = true
+tracing_sampling_rate_per_million = 1000000
+```
+
+To configure CRI-O to use conmon-rs, you require at least the latest CRI-O
+v1.25.x and conmon-rs v0.4.0. Then a configuration drop-in like this can be used
+to make CRI-O use conmon-rs:
+
+```shell
+> cat /etc/crio/crio.conf.d/99-runtimes.conf
```

Comment:
```suggestion
cat /etc/crio/crio.conf.d/99-runtimes.conf
```
or omit the `cat`, and tell people that this is an example file that could go into `/etc/crio/crio.conf.d`.